### PR TITLE
45033746 - Project 7: Siamese CNN Classifier for Alzheimers Disease

### PR DESCRIPTION
**Pull request by tweeeb**
## Description

This is a pull request to merge the recognition problem for project 7: Siamese CNN binary classifier for Alzheimer's disease using the ADNI dataset. The model comprises a triple Siamese convolutional neural network and a binary classifier. The model is trained to classify 2D images slices of brain scans as Alzheimer's disease (AD) and normal cognitive (NC).

- Includes README.md with documentation
- Models can be generated and trained with train,py
- Models can be tested for accuracy with predict.py
- Loss graphs are saved to ./assets and can be viewed after running train.py

## Requirements
- topic-recognition branch
- commit log history
- train.py must be run before predict.py

Please note, the data is retrieved via the rangpur location if on a linux system, and is retrieved via the file structure nominated in the readme on a windows system. Adjustments can be made to the filepath at the top of the dataset.py file if this is not the configuration preferred.

Requires PyTorch 2.0.1 and matplotlib 3.7.2

45033746
Treffery Webb (@tweeeb)

### CODE REVIEW

**Comment by nathasha-naranpanawa** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme: Poor
- details on algorithm and usage provided
- no information on training choices
- **no results/figures/example predictions shown**

Commit messages: Good

Code: 
- good design
- well commented
- algorithm/architectures seem correct

Functionality/Performance: 
- **no numbers/plots provided to judge performance**

General comments:
- algorithm solves the problem apporpriately, **with unknown performance**

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem, performance unknow no testing results -3
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage, no result output, no training plots -3
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe OK, no refs, no plots -2, could have problem info etc. -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required, remove .idea files etc. -2
Request Description good

**Comment by shakes76** [comment]
Feedback marks possible +2 if the requested changes are made (see above).

**Comment by wangzhaomxy** [comment]
No feedback attempt and no feedback marks granted.

**Comment by tweeeb** [comment]
@wangzhaomxy the edstem post specified no hard deadline, why did that change???

**Comment by wangzhaomxy** [comment]
> @wangzhaomxy the edstem post specified no hard deadline, why did that change???

I just marked all problem pull requests and was making sure I had covered everyone.  You can still get marks after finishing the adjustment. I will get another inspection before the due date. However, we cannot wait for endless time, we have to update your grades to the school in a few days.  Please adjust it ASAP.

**Comment by tweeeb** [comment]
@wangzhaomxy 

> > @wangzhaomxy the edstem post specified no hard deadline, why did that change???
> 
> I just marked all problem pull requests and was making sure I had covered everyone. You can still get marks after finishing the adjustment. I will get another inspection before the due date. However, we cannot wait for endless time, we have to update your grades to the school in a few days. Please adjust it ASAP.

Thank you! I got the final feedback only two days ago while on a buisness trip 😭😭 will update ASAP
